packages/frontend/src/components/UnderlyingData/UnderlyingDataResultsTable.tsx

```diff
@@ -28,6 +30,17 @@ const UnderlyingDataResultsTable: FC<{
         columnHeader,
     });
 
+    if (resultsData === undefined) {
```

Comment:
This is much less reliable than checking if the underlying query is loading. 

You could do this by passing the resulting react query hook into the prop (or accessing the hook directly) then checking the `isLoading` prop